modules/distribution-zones/src/main/java/org/apache/ignite/internal/distributionzones/DistributionZoneManager.java

```diff
@@ -197,12 +235,114 @@ public CompletableFuture<Void> dropZone(String name) {
     /** {@inheritDoc} */
     @Override
     public void start() {
-
+        zonesConfiguration.distributionZones().listenElements(new ZonesConfigurationListener());
     }
 
     /** {@inheritDoc} */
     @Override
     public void stop() throws Exception {
 
     }
+
+    private class ZonesConfigurationListener implements ConfigurationNamedListListener<DistributionZoneView> {
+        @Override
+        public CompletableFuture<?> onCreate(ConfigurationNotificationEvent<DistributionZoneView> ctx) {
+            updateMetaStorageOnZoneCreateOrUpdate(ctx.newValue().zoneId(), ctx.storageRevision());
+
+            return completedFuture(null);
+        }
+
+        @Override
+        public CompletableFuture<?> onDelete(ConfigurationNotificationEvent<DistributionZoneView> ctx) {
+            updateMetaStorageOnZoneDelete(ctx.oldValue().zoneId(), ctx.storageRevision());
+
+            return completedFuture(null);
+        }
+
+        @Override
+        public CompletableFuture<?> onUpdate(ConfigurationNotificationEvent<DistributionZoneView> ctx) {
+            updateMetaStorageOnZoneCreateOrUpdate(ctx.newValue().zoneId(), ctx.storageRevision());
```

Comment:
That'll be a bit incorrect, in case of zone updates it's requited to recalculate dataNodes only within following scenarios:

- In case of filter change.
- In case of timer changes if there are active timers. (re-scheduling)
In all other cases there's no need to recalculate dataNodes, however it's required to invoke triggerRevision immediately.

Reply:
As far as I can understand, if got here, that means that any of `dataNodesAutoAdjust()`, `dataNodesAutoAdjustScaleUp()`, `dataNodesAutoAdjustScaleDown()` has been changed, so there is no "other cases". Case with renaming will be handled by separate handler `onRename`